### DISCUSSION TITLE
RM Kaltstart - Kein USB remount

### DISCUSSION THREAD

**Discussion by MaHe801310** [OpenCCU/OpenCCU#2371] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Ein bereits gemounteter USB Stick wird nach einem Zentralenkaltstart (zielausgelöst od. Spannungsverlust)
nicht wieder erneut eingebunden.

Erst ein mechanisch-manuelles Trennen und Wiederverbinden macht Selbigen der Zentrale wieder zugänglich.

Nachvollziehbar unter Systemsteuerung => Allgemeine Einstellungen, jeweils vor und nach dem Kaltstart.

Auch das permanente mounten mittels rc.postinit / rc.prelocal funktioniert trotz korrekter Umsetzung nicht.

**Frage: Ist das gewollt, dass das /mnt-Verzeichnis schreibgeschützt ist ?**

Nur die erwähnte Vorgehensweise bindet den USB-Stick korrekt wieder ein.

### Describe the behavior you expected

Auch nach einem Zentralenkaltneustart mittels herunterfahren oder Spannungsverlust und erneuter Spannungszufuhr
sollte der verwendete USB-Stick automatisch wieder eingebunden werden.

### Steps to reproduce the issue

1. USB-Stick anstecken. 
2. Kontrolle => Systemsteuerung, Allgemeine Einstellungen  => vorhanden.
3. Spannungsverlust herbeiführen (PowerOff, regulär od. forciert).
4. Spannungsversorgung wiederherstellen
5. Kontrolle => Systemsteuerung, Allgemeine Einstellungen  => nicht vorhanden
6. mechanisch-manuelles Steckwechseln (ab / an) des USB-Sticks.
7. Kontrolle => Systemsteuerung, Allgemeine Einstellungen  => vorhanden.
...

### What is the version this bug report is based on?

3.69.7.20230506

### Which base platform are you running?

rpi3 (RaspberryPi3)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
N/A
```

### Additional information

In Verwendung ein Intenso UltraLine 16GB USB-Stick, sowie weitere. => Gleiche Symptome.

**Comment by jp112sdl** [comment]
Was zeigt denn `lsusb` nach einem Neustart, wenn der Stick (in der WebUI) nicht angezeigt/erkannt wird?

Ich hatte sowas Merkwürdiges selbst schon mal (außerhalb von RaspberryMatic), da lag es an der Stromversorgung (Kabel vom Netzteil zum Pi). Irgendwas muss beim Booten kurzzeitig die Versorgung in Knie gezwungen haben, was die USB Geräte nicht toll fanden.
Ging _immer_... auf einmal nicht mehr. Nix verändert und so... der Klassiker ^^
Neues USB-Kabel ran - alles wieder schick.

**Reply by MaHe801310** [reply]
root@ccu3-webui:~# lsusb
Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 003: ID 0424:2514
Bus 001 Device 002: ID 0424:2514
Bus 001 Device 004: ID 0424:7800

Es kann natürlich sein, dass das verwendete Keystone-USB - Modul nicht kompatibel ist.
Bei Gelegenheit werde ich dies nochmals mit einem anderen Kabel bzw. "Nur-Kabel" prüfen.

Netzteil habe ich schon auf Verdacht getauscht, (Original Pi / USB-C an Adapte) jedoch mit gleichem Ergebnis.

Hier nochmal ein lsusb nach Ab- und Wiedereinstecken des besagten Sticks.

root@ccu3-webui:~# lsusb
Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 003: ID 0424:2514
Bus 001 Device 002: ID 0424:2514
Bus 001 Device 004: ID 0424:7800
Bus 001 Device 005: ID 058f:6387

Nahversuch (ohne Kabel) noch keinen, kommt aber wegen der Einbauhöhe der CCU3 nicht in Frage.

Ich denke, dass ich dann doch auf ein FRITZ!Nas Volume ausweichen werde.

Vielen Dank rückwirkend allen Beteiligten schonmal.

**Comment by Baxxy13** [comment]
Ein generelles Problem ist das eher weniger. 
Sonst würden sich sicher viel mehr Pi3B/USB-Stick Nutzer melden. 

Ich tippe auch drauf das der Stick beim Booten nicht korrekt initialisiert/erkannt wird und daher gar nicht zur Verfügung steht. Dann kann der natürlich auch nicht gemountet werden. 

In irgendeinem Logfile kann man nach dem Booten auch "nachlesen" welche USB-Devices registriert wurden, weiß aber gerade nicht genau wo.

**Reply by jens-maus** [reply]
Ich gehe auch nicht von einem generellen Problem hier aus und habe das daher mal in die Discussions verschoben...

**Comment by Baxxy13** [comment, marked as answer]
So, nun habe ich nen Pi3B samt RPI-RF-MOD und einen USB-Stick aus der "Testgeräte-Kiste" gekramt und das mal mit RM 3.69.7.20230626 durchexerziert.

Zusammengefasst: keine Probleme.

Ob bei einem Zentralen-Reboot oder aber längerem "Poweroff" (samt entfernen der Stromzufuhr), der Stick wird immer korrekt erkannt und gemounted.

```
Jul 24 16:29:44 localhost kern.info kernel: [    3.738751] usb 1-1.2: new high-speed USB device number 4 using dwc_otg
Jul 24 16:29:44 localhost kern.info kernel: [    3.840294] usb 1-1.2: New USB device found, idVendor=0781, idProduct=5583, bcdDevice= 1.00
Jul 24 16:29:44 localhost kern.info kernel: [    3.840325] usb 1-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
Jul 24 16:29:44 localhost kern.info kernel: [    3.840349] usb 1-1.2: Product: Ultra Fit
Jul 24 16:29:44 localhost kern.info kernel: [    3.840370] usb 1-1.2: Manufacturer: SanDisk
Jul 24 16:29:44 localhost kern.info kernel: [    3.840390] usb 1-1.2: SerialNumber: 4C530000021022102195
Jul 24 16:29:44 localhost kern.info kernel: [    3.841718] usb-storage 1-1.2:1.0: USB Mass Storage device detected
Jul 24 16:29:44 localhost kern.info kernel: [    3.842971] scsi host0: usb-storage 1-1.2:1.0
Jul 24 16:29:44 localhost kern.notice kernel: [    4.864432] scsi 0:0:0:0: Direct-Access     SanDisk  Ultra Fit        1.00 PQ: 0 ANSI: 6
Jul 24 16:29:44 localhost kern.notice kernel: [    4.866851] sd 0:0:0:0: [sda] 30031872 512-byte logical blocks: (15.4 GB/14.3 GiB)
Jul 24 16:29:44 localhost kern.notice kernel: [    4.867952] sd 0:0:0:0: [sda] Write Protect is off
Jul 24 16:29:44 localhost kern.debug kernel: [    4.867980] sd 0:0:0:0: [sda] Mode Sense: 43 00 00 00
Jul 24 16:29:44 localhost kern.notice kernel: [    4.868670] sd 0:0:0:0: [sda] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
Jul 24 16:29:44 localhost kern.info kernel: [    4.881486]  sda: sda1
Jul 24 16:29:44 localhost kern.notice kernel: [    4.882089] sd 0:0:0:0: [sda] Attached SCSI removable disk
```
